src/lib/zf/private/reactor.c

```diff
@@ -508,10 +508,11 @@ __zf_reactor_perform(struct zf_stack* st, unsigned spin_cnt)
            * from the future in the next RX buffer. */
           pkt_id next_packet_id = 0;
           char* next_packet;
-          bool is_efct = vi->nic_type.arch == EF_VI_ARCH_EFCT;
+          bool is_rx_ref = *zf_stack_res_nic_flags(st, nic) &
```

Comment:
You're using these flags on the critical path. The `zf_stack_impl` is used for cold path state. If we need the checks on the hot path that information should move to `zf_stack_nic`. That might just be the RX_REF capability - probably needs a bit of care to ensure that this is done in a way that doesn't impact performance.